server/aws-lsp-codewhisperer/src/language-server/agenticChat/agenticChatController.ts

```diff
@@ -425,9 +426,18 @@ export class AgenticChatController implements ChatHandlers {
                     content: [toolResultContent],
                 })
 
-                // no need to write tool result for listDir and fsRead into chat stream
-                if (toolUse.name !== 'fsRead' && toolUse.name !== 'listDirectory') {
-                    await chatResultStream.writeResultBlock({ body: toolResultMessage(toolUse, result) })
+                switch (toolUse.name) {
```

Comment:
For the sake of future MCP support, I think this tool specific branching and logic should be isolated from the core logic.